### DISCUSSION TITLE
When users try to accept an invitation to a private repo, it fails with "This invitation is invalid"

### DISCUSSION THREAD

**Discussion by dkershaw10** [community/community#67576] [Other Feature Feedback, Questions, & Ideas]
I have a private repo that I'm trying to share with 30 or so "Outside Collaborators". The repo is part of an organization, but not sure that this should make a difference.  When I send out invites (by adding folks via the `Collaborator and teams` settings page), some Outside Collaborators are able to accept and join, and others get a "This invitation is invalid" failure, even though the invitation has not expired.  In some cases, re-adding/re-inviting outside collaborators and having them accept a fresh invitation request immediately works (and they are able to join the private repo), but for some others, no matter what I try, they are unable to join.

Does anyone have any idea what the problem could be here?  It really shouldn't be this hard.

Any help would be greatly appreciated, otherwise I will have to resort to making the repo public, which I'm really not ready to do at this stage.

**Comment by jeffwilcox** [comment]
Is it possible that you were an administrator of the repository when you extended the grant, but at some point not an admin, during the 7 day period an invite is valid for typically?

As more companies develop just-in-time solutions to rally around the concept of "least privileged access", this means that in many cases, none of us are long-term admins at all of a repo... so if I'm an admin for an hour, changing some settings, or inviting someone, any invitation will be canceled when I downgrade my access an hour later.

Just a thought, maybe something like that is impacting you?

**Reply by jeffwilcox** [reply]
[Update: just realized that the OP works at my employer and is definitely working through the least privilege access system we have; I will reach out internally with information on how we enable this scenario]

**Reply by natke** [reply]
@jeffwilcox I am facing this issue too (and work at MSFT and use just in time elevation). Is the workaround to add the external collaborator from within the repo management system rather than in GitHub?

**Reply by jeffwilcox** [reply]
Yes natke, that portal invite will use the GitHub REST API behind the scenes to send an invite that will outlast your elevation.

**Reply by natke** [reply]
Thanks! I assume I would need to revoke (from within GitHub) the ones that didn't work? There doesn't seem to be a way of revoking from the repo management system

**Reply by qaisjp** [reply]
At Stripe, we're all of a sudden now seeing this for invites (invites to private repos) created by our GitHub App. Any idea on how you fixed the "This invitation is invalid." problem?

edit: it was a temporary issue, we contacted GitHub and they fixed it

**Reply by bobbytables** [reply]
I'm also seeing this from the perspective of a user. I can view the "You've been invited page" just fine, but when I click "Accept Invite" I get the error: "This invitation is invalid" – I didnt hit this a few days ago from the same flow.

**Reply by meorphis** [reply]
We have also been running into this issue with our GitHub app at Stainless.

**Comment by supuna97** [comment]
Are you a administrator of this repo? If you are a administrator of this repo, you can send invitations for outside collaborators. You check your role of this repo. And try it again with administrator.
